src/nvim/testdir/shared.vim

```diff
@@ -196,16 +196,26 @@ func s:feedkeys(timer)
   call feedkeys('x', 'nt')
 endfunc
 
+" Get $VIMPROG to run Vim executable.
+" The Makefile writes it as the first line in the "vimcmd" file.
+" Nvim: uses $NVIM_TEST_ARG0.
+func GetVimProg()
+  if empty($NVIM_TEST_ARG0)
+    " Assume the script was sourced instead of running "make".
+    return '../../../build/bin/nvim'
```

Comment:
could use v:progpath then

Reply:
This tries to follow the (not really clear) use case mentioned in https://github.com/vim/vim/pull/4806#issuecomment-520847278.